src/utils/vector-db/manager.ts

```diff
@@ -48,48 +52,58 @@ export class VectorDbManager {
       chunkSize: number
-      overwrite?: boolean
+      reindexAll?: boolean
     },
     updateProgress?: (indexProgress: IndexProgress) => void,
   ): Promise<void> {
     let filesToIndex: TFile[]
-    if (options.overwrite) {
+    if (options.reindexAll) {
       filesToIndex = this.app.vault.getMarkdownFiles()
       await this.repository.clearAllVectors(embeddingModel)
     } else {
-      const indexedFilePaths =
-        await this.repository.getIndexedFilePaths(embeddingModel)
-      for (const filePath of indexedFilePaths) {
-        if (!this.app.vault.getAbstractFileByPath(filePath)) {
-          await this.repository.deleteVectorsForMultipleFiles(
-            [filePath],
-            embeddingModel,
-          )
+      const deleteVectorsForDeletedFiles = async () => {
+        const indexedFilePaths =
+          await this.repository.getIndexedFilePaths(embeddingModel)
+        for (const filePath of indexedFilePaths) {
+          if (!this.app.vault.getAbstractFileByPath(filePath)) {
+            await this.repository.deleteVectorsForMultipleFiles(
+              [filePath],
+              embeddingModel,
+            )
+          }
         }
       }
-      const markdownFiles = this.app.vault.getMarkdownFiles()
 
-      filesToIndex = await Promise.all(
-        markdownFiles.map(async (file) => {
-          const fileChunks = await this.repository.getVectorsByFilePath(
-            file.path,
-            embeddingModel,
-          )
-          if (fileChunks.length === 0) {
-            const fileContent = await this.app.vault.cachedRead(file)
-            if (fileContent.length === 0) {
-              return null
+      const getFilesToIndex = async (): Promise<TFile[]> => {
+        const markdownFiles = this.app.vault.getMarkdownFiles()
+        const filesToIndex = await Promise.all(
+          markdownFiles.map(async (file) => {
+            const fileChunks = await this.repository.getVectorsByFilePath(
+              file.path,
+              embeddingModel,
+            )
+            if (fileChunks.length === 0) {
+              // File is not indexed, so we need to index it
+              const fileContent = await this.app.vault.cachedRead(file)
+              if (fileContent.length === 0) {
+                // Ignore empty files
+                return null
+              }
+              return file
             }
-            return file
-          }
-          const outOfDate = file.stat.mtime > fileChunks[0].mtime
-          if (outOfDate) {
-            return file
-          }
-          return null
-        }),
-      ).then((files) => files.filter(Boolean) as TFile[])
+            const outOfDate = file.stat.mtime > fileChunks[0].mtime
+            if (outOfDate) {
+              // File has changed, so we need to re-index it
+              return file
+            }
+            return null
+          }),
+        ).then((files) => files.filter(Boolean) as TFile[])
+        return filesToIndex
+      }
```

Comment:
Please move these functions to separate private functions to improve readability.